qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -93,6 +132,270 @@ def __init__(self, control_config: Dict[Tuple[int, ...], List[ControlChannel]] =
+                map will be updated.
+            group: The calibration group from which to draw the parameters. If not specified
+                this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+            inst_map: The instruction schedule map to update. If None is given then the default
+                instruction schedule map (i.e. self._inst_map) will be updated.
+        """
+        inst_map = inst_map or self._inst_map
+
+        for key in self._schedules:
+            sched_name = key.schedule
+
+            if schedules is not None and sched_name not in schedules:
+                continue
+
+            if qubits is not None:
+                inst_map.add(
+                    instruction=sched_name,
+                    qubits=qubits,
+                    schedule=self.get_schedule(
+                        sched_name, qubits, group=group, cutoff_date=cutoff_date
+                    ),
+                )
+
+            else:
+                for qubits_ in self.operated_qubits[self._schedules_qubits[key]]:
+                    try:
+                        inst_map.add(
+                            instruction=sched_name,
+                            qubits=qubits_,
+                            schedule=self.get_schedule(
+                                sched_name, qubits_, group=group, cutoff_date=cutoff_date
+                            ),
+                        )
+                    except CalibrationError:
+                        # get_schedule may raise an error if not all parameters have values or
+                        # default values. In this case we ignore and continue updating inst_map.
+                        pass
+
+    def inst_map_add(
+        self,
+        instruction_name: str,
+        qubits: Tuple[int],
+        schedule_name: Optional[str] = None,
+        assign_params: Optional[Dict[Union[str, ParameterKey], ParameterValueType]] = None,
+    ):
+        """Update a single instruction in the instruction schedule map.
+
+        This method can be used to update a single instruction for the given qubits but
+        it can also be used by experiments that define custom gates with parameters
+        such as the :class:`Rabi` experiment. In a Rabi experiment there is a gate named
+        "Rabi" that scans a pulse with a custom amplitude. Therefore we would do
+
+        .. code-block:: python
+
+            cals.inst_map_add("Rabi", (0, ), "xp", assign_params={"amp": Parameter("amp")})
+
+        to temporarily add a pulse for the Rabi gate in the instruction schedule map. This
+        then allows calling :code:`transpile(circ, inst_map=cals.instruction_schedule_map)`.
```

Comment:
```suggestion
        then allows calling :code:`transpile(circ, inst_map=cals.instruction_schedule_map)`.
```
This method doesn't exist

Reply:
Fixed: a8a186c